src/SparqlLuaQueryRunner.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\SPARQL;
+
+use MediaWiki\Http\HttpRequestFactory;
+
+class SparqlLuaQueryRunner {
+
+	private HttpRequestFactory $requestFactory;
+
+	/**
+	 * @param HttpRequestFactory $requestFactory
+	 */
+	public function __construct( HttpRequestFactory $requestFactory ) {
+		$this->requestFactory = $requestFactory;
+	}
+
+	/**
+	 * @param string $sparql
+	 *
+	 * @return array<mixed>
+	 */
+	public function runQuery( string $sparql ): array {
+		global $wgSPARQLEndpoint;
```

Comment:
This should be injected in the constructor. Global access is forbidden past the dependency graph construction in our entry points.